Dockerfile

```diff
@@ -1,11 +1,20 @@
 FROM dmscid/lewis-depends:latest
 
+# Copying this separately from the rest of lewis allows the
+# pip install step to be cached until the requirements change.
+# Rebuilding is faster and doesn't require an internet connection.
+COPY requirements.txt /lewis/
+
+# Install any missing PIP requirements, clear PIP cache, and delete all compiled Python objects
+RUN pip install -r /lewis/requirements.txt && \
+    rm -rf /root/.cache/pip/* && \
+    find \( -name '*.pyc' -o -name '*.pyo' \) -exec rm -rf '{}' +
+
 # Lewis source is pulled in from current directory (note .dockerignore filters out some files)
 COPY . /lewis
 
-# Install any missing PIP requirements, clear PIP cache, and delete all compiled Python objects
+# Install lewis in-place and remove compiled Python objects
 RUN pip install -e /lewis && \
```

Comment:
I definitely like `pip install -e /lewis` better than a normal install.

This way, you can `docker exec` into a running container, or just start with `docker run -it --entrypoint /init.sh lewis`, and edit/debug/execute lewis in its own little sandbox environment that has all the dependencies pre-installed. This is particularly nice if you can't or don't want to deal with installing EPICS on the host OS but want to make or test some quick changes to lewis.

The main advantage I saw to using `pip install -e` over the old method was that it got rid of the need for `requirements.txt` to exist, which meant that the dependencies would only be listed in one place: the setup file. But if we're keeping `requirements.txt` anyway, is there much point to using `pip install -e` inside the container? 

I can think of only two effects it has:
- You can now execute `$ lewis` and `$ lewis-control` directly, instead of `$ /lewis/lewis.py` and `$ /lewis/lewis-control.py` (hidden most of the time since it's the entrypoint)
- `setup.py` may install something that `requirements.txt` didn't include (which would technically be a bug since they should be the same?)

The difference seems quite minimal. The first point won't be noticeable when just starting a lewis simulation normally. The second point is probably a disadvantage, since it could hide bugs.

If I remember right, the main reason for `lewis-depends` was that some dependencies (pcaspy and pyzmq) required build tools and various other `-dev` packages in order to install. This is pretty heavyweight and time-consuming to download and install, so we didn't want it to happen every time the `lewis` image is rebuilt. Right now, `lewis-depends` explicitly installs just those two packages and isn't aware of `requirements.txt` or any part of lewis really.

We could change that and download and `pip install -r` the latest `requirements.txt` from the master branch of `lewis`... but this means `lewis-depends` would have to be rebuilt every time `requirements.txt` changes. Or, at least, whenever we want to stop having to download/install recently added `requirements` every time `lewis` is rebuilt.